seqr/views/apis/dashboard_api_tests.py

```diff
@@ -120,3 +122,116 @@ def test_export_projects_table(self):
         response = self.client.get('{}?file_format=csv'.format(url))
         self.assertEqual(response.status_code, 400)
         self.assertDictEqual(response.json(), {'error': 'Invalid file_format: csv'})
+
+
+# Tests for AnVIL access disabled
+class LocalDashboardPageTest(AuthenticationTestCase, DashboardPageTest):
+    fixtures = ['users', '1kg_project']
+
+
+# Test for permissions from AnVIL only
+@mock.patch('seqr.views.utils.terra_api_utils.TERRA_API_ROOT_URL', TEST_TERRA_API_ROOT_URL)
+class AnvilDashboardPageTest(AnvilAuthenticationTestCase, DashboardPageTest):
+    fixtures = ['users', 'social_auth_data', '1kg_project']
+
+    def setUp(self):
+        patcher = mock.patch('seqr.views.utils.permissions_utils.list_anvil_workspaces')
+        patcher.start().side_effect = get_workspaces_side_effect
+        self.addCleanup(patcher.stop)
+        patcher = mock.patch('seqr.views.utils.terra_api_utils._service_account_session')
+        patcher.start().get.side_effect = get_ws_acl_side_effect
+        self.addCleanup(patcher.stop)
+        patcher = mock.patch('seqr.views.utils.terra_api_utils.time')
+        patcher.start().return_value = 1603287741 + 10
+        self.addCleanup(patcher.stop)
+
+
+MIX_TEST_WORKSPACES = [
+        {
+            'public': False,
+            'accessLevel': 'READER',
+            'workspace': {
+                'namespace': 'my-seqr-billing',
+                'name': 'anvil-project 1000 Genomes Demo'
+            }
+        },
+    ]
+
+MIX_TEST_ALCS = {'acl': {
+        'test_user_manager@test.com': {
+            "accessLevel": "WRITER",
+            "pending": False,
+            "canShare": True,
+            "canCompute": True
+        },
+        'test_user_no_staff@test.com': {
+            "accessLevel": "READER",
+            "pending": False,
+            "canShare": True,
+            "canCompute": True
+        }
+    }
+}
+
+
+@mock.patch('seqr.views.utils.terra_api_utils.TERRA_API_ROOT_URL', TEST_TERRA_API_ROOT_URL)
+class MixDashboardPageTest(AuthenticationTestCase):
+    fixtures = ['users', 'social_auth_data', '1kg_project']
+
+    @mock.patch('seqr.views.utils.permissions_utils.list_anvil_workspaces')
+    @mock.patch('seqr.views.utils.terra_api_utils._service_account_session')
+    @mock.patch('seqr.views.utils.terra_api_utils.time')
+    def test_dashboard_page_data(self, mock_time, mock_service_account_session, mock_list_anvil_workspaces):
+        # Users can see the projects that AnVIL allows
+        url = reverse(dashboard_page_data)
+        self.login_collaborator()
+        mock_time.time.return_value = 1603287741 + 10
+        mock_service_account_session.get.return_value = MIX_TEST_ALCS
+        mock_list_anvil_workspaces.return_value = MIX_TEST_WORKSPACES
```

Comment:
Would there be a way to do the mixed tests in a way the reused the same terra mocking as the pure anvil test? When the data coming back from AnVIL is the same, it makes it really clear what is different in the Mix case vs the pure anvil case. See my comment on the `AnvilAuthenticationTestCase` about using helper methods to make it easier to test that there is different expected data

Reply:
Yes. I would try to define a new mix-authentication class and define different mocking and expected results in it.

Reply:
Hmmm I see what you are saying, but I was actually saying "what if the anvil permissions were the same in the anvil only and mix case". Then it would be really obvious to the reader what is different in the results between them, whereas if the permissions coming back from AnVIL are different its really hard to tell

Reply:
The problem is that there will be no difference between local and mix if we use the same mocks as anvil only. It is hard to figure out if the anvil configurations have taken effect. Unless I make a slight change to the fixture data for the projects. For example, add `disable_staff_access` to one of the projects.
By the way, the new mix-auth class seems necessary because it enables both local and anvil permissions which won't happen in either local or anvil authentication classes.